### PR TITLE
chore: Concatenate AMI IDs in logs

### PR DESCRIPTION
**Pull request by engedaam**
Fixes # 

**Description**

- When user specify amiSelector to `aws::name: "*"` karpenter logs will become filled with all the AMI IDs that are discovered by ec2.

- `2023-04-14T16:56:32.980Z	DEBUG	controller.node	discovered images	{"commit": "b4f1d60-dirty", "node": "ip-192-168-112-220.us-west-2.compute.internal", "ids": "ami-0789036840ebc8c25, ami-0123d5874c000c92e, ami-01e2c4303a7750e9f, ami-0438e722aaf2037af, ami-0696f65e1894f23c1, ami-06d0b3b36f79ffd17, ami-0386da4fafcdb3286, ami-0f4af98dc9062fc55, ami-055613971ee89acd7, ami-06055064d6004521f, ami-065a7f1c72ec3164b, ami-05692094f77d6c828, ami-0fe99b168ad207a9e, ami-07d7dd2821eadc9b7, ami-093107ba06ec290c8, ami-019f57bd424d44fd3, ami-01fc410639698b4ca, ami-0ccdef9b637fc187c, ami-0e8c547337376a477, ami-0d44c92d6423bd06b, ami-01042020b7eb94eb7, ami-0fe0482fe8b6a011d, ami-05d00b2a498b88054, ami-0a86c6d5ab23eefa7, ami-0525d0d0a99a22dbb and 11778 other(s)", "count": 11803}`

**How was this change tested?**

* Manually Tested 

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/providers/amifamily/ami.go [hunk 1]
```diff
@@ -219,11 +219,22 @@ func (p *Provider) fetchAMIsFromEC2(ctx context.Context, amiSelector map[string]
...
+		logging.FromContext(ctx).With("ami-ids", concatenateAmiIDs(amiIDs)).Debugf("discovered images")
 	}
 	return output.Images, nil
 }
 
+func concatenateAmiIDs(amisIds []string) string {
+	var amisIdsString string
+	if len(amisIds) > 5 {
+		amisIdsString = strings.Join(amisIds[:5], ", ")
+		amisIdsString += fmt.Sprintf(", (%d more amis)", len(amisIds)-5)
```
Comment (jonathan-innis) [inline pkg/providers/amifamily/ami.go:231]: Can we align the formatting of this with [instanceTypeList](https://github.com/aws/karpenter-core/pull/273/files#diff-f0cdb8c0527c28b9b19283810ab0deadc411c48d0f66ef2255109dd7824f7e7eR356)

File: pkg/providers/amifamily/ami.go [hunk 2]
```diff
@@ -219,11 +219,22 @@ func (p *Provider) fetchAMIsFromEC2(ctx context.Context, amiSelector map[string]
 	p.ec2Cache.SetDefault(fmt.Sprint(hash), output.Images)
 	amiIDs := lo.Map(output.Images, func(ami *ec2.Image, _ int) string { return *ami.ImageId })
 	if p.cm.HasChanged("amiIDs", amiIDs) {
-		logging.FromContext(ctx).With("ami-ids", amiIDs).Debugf("discovered images")
+		logging.FromContext(ctx).With("ami-ids", concatenateAmiIDs(amiIDs)).Debugf("discovered images")
```
Comment (jonathan-innis) [inline pkg/providers/amifamily/ami.go:222]:
```suggestion
		logging.FromContext(ctx).With("ids", concatenateAmiIDs(amiIDs), "count", len(amiIDs)).Debugf("discovered images")
```

We've already scoped this to "images" so I think just saying "ids" is fine. Also, I think making "count" a structured value is probably useful here as well

File: pkg/providers/amifamily/ami.go [hunk 3]
```diff
@@ -219,11 +219,22 @@ func (p *Provider) fetchAMIsFromEC2(ctx context.Context, amiSelector map[string]
...
 	}
 	return output.Images, nil
 }
 
+func concatenateAmiIDs(amisIds []string) string {
+	var amisIdsString string
```
Comment (jonathan-innis) [inline pkg/providers/amifamily/ami.go:228]:
```suggestion
	var amiString string
```

nit: It's easier to read the variable name this way.

**Review by jonathan-innis — COMMENTED** [review]
Nice! 🎉

File: pkg/providers/amifamily/ami.go [hunk 4]
```diff
@@ -219,11 +219,22 @@ func (p *Provider) fetchAMIsFromEC2(ctx context.Context, amiSelector map[string]
...
 	if p.cm.HasChanged("amiIDs", amiIDs) {
-		logging.FromContext(ctx).With("ami-ids", amiIDs).Debugf("discovered images")
+		logging.FromContext(ctx).With("ids", concatenateAmiIDs(amiIDs), "count", len(amiIDs)).Debugf("discovered images")
 	}
 	return output.Images, nil
 }
 
+func concatenateAmiIDs(amisIds []string) string {
+	var amiString string
+	if len(amisIds) > 5 {
```
Comment (njtran) [inline pkg/providers/amifamily/ami.go:229]: IMO I think there are plenty of valid use cases to have more than 5 distinct AMIs that are used. Can we stick with 25 here and reduce it later if we feel like it's still too noisy?

**Review by njtran — APPROVED** [review]

File: pkg/providers/amifamily/ami.go (hunk 4 above)
Comment (jonathan-innis) [inline pkg/providers/amifamily/ami.go:229]:
```suggestion
	if len(amisIds) > 25 {
```

**Review by jonathan-innis — CHANGES_REQUESTED** [review]

**Review by jonathan-innis — APPROVED** [review]
Ship it! 🎉

**Pull request merged**